test/integration/core/kernel/Factory/ResourceFactoryTest.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace oat\generis\test\integration\core\kernel\Factory;
+
+use core_kernel_classes_Property;
+use core_kernel_classes_Resource;
+use LogicException;
+use oat\generis\model\kernel\Factory\ResourceFactory;
+use oat\generis\test\TestCase;
+
+class ResourceFactoryTest extends TestCase
+{
+    /** @var ResourceFactory */
+    private $resourceFactory;
+
+    public function setUp()
+    {
+        $this->resourceFactory = new ResourceFactory();
+    }
+
+    /**
+     * @dataProvider fqcnDataProvider
+     * @param string $fqcn
+     */
+    public function testCreateMethodReturnsTheCorrectInstance($fqcn)
+    {
+        $this->assertInstanceOf(core_kernel_classes_Resource::class, $this->resourceFactory->create(
```

Comment:
Also, since you are using different class instances in the dataprovider (which os a good approach), you should assert on the actual instance, not on the parent. You expect the child classes, not the generic resource class every time

Reply:
True, forgot to refact this part :)